src/Stream/PartitionFilter.h

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#pragma once
+
+#include "KafkaW/PollStatus.h"
```

Comment:
Because _PollStatus.h_ only defines an enum class declaration and _PartitionFilter.h_ is only included in one location, I consider this optimisation to not be worth the effort. I will still do it if you insist though.

Reply:
I prefer to establish the habit to
> Never #include a header when a forward declaration will suffice.

but I won't insist.
(Guideline from https://herbsutter.com/2013/08/19/gotw-7a-solution-minimizing-compile-time-dependencies-part-1/)

Reply:
+1 for doing this by habit

Reply:
There is such a thing as "cognitive load", and making this change adds to that by adding code for no performance benefit (compile time or otherwise). Seeing as I have both of you against me though, I will make the change.